lab-lee/app/component/landing/login/login.html

```diff
@@ -0,0 +1,45 @@
+  <section class="login-form">
+  <form
+    name="loginForm"
+    ng-submit="loginCtrl.login()" novalidate>
+    <div
+      ng-class="{
+        'has-error': loginForm.username.$invalid,
+        'has-success': loginForm.username.$valid,
+      }"
+      class="form-group">
+      <input type="text"
+        name="username"
+        uib-tooltip="Username must have 5 characters"
+        tooltip-enable="loginForm.$submitted && loginForm.username.$invalid"
+        tooltip-is-open="loginForm.$submitted && loginForm.username.$invalid &&
+        loginForm.username.$focus"
+        tooltip-placement="top-left"
+        tooltip-trigger="'focus'"
+        placeholder="Username"
+        class="form-control"
+        required ng-minlength="5"
+        ng-model="loginCtrl.user.username">
+
+    <div
+      ng-class="{
+        'has-error': loginForm.password.$invalid && loginForm.$submitted,
+        'has-success': loginForm.password.$valid,
+      }"
+      class="form-group">
+      <input type="password" name="password"
+        ng-disabled="loginForm.username.$invalid"
+        uib-tooltip="Password must have 8 charicters"
```

Comment:
Good use of uib-tooltip, but you should type it out yourself next time instead of copying Duncan's code to gain that muscle memory (same typo :P).